core/modules/cart/components/AddToCart.ts

```diff
@@ -20,13 +21,31 @@ export const AddToCart = {
   methods: {
     async addToCart (product: Product) {
       this.isAddingToCart = true
-      const diffLog = await this.$store.dispatch('cart/addItem', { productToAdd: product })
-      if (diffLog.clientNotifications && diffLog.clientNotifications.length > 0) {
-        diffLog.clientNotifications.forEach(notificationData => {
-          this.$store.dispatch('notification/spawnNotification', notificationData, { root: true })
+      try {
+        const diffLog = await this.$store.dispatch('cart/addItem', { productToAdd: product })
+        if (diffLog) {
+          if (diffLog.clientNotifications && diffLog.clientNotifications.length > 0) {
+            diffLog.clientNotifications.forEach(notificationData => {
+              this.$store.dispatch('notification/spawnNotification', notificationData, { root: true })
+            })
+          }
+        } else {
+          this.$store.dispatch('notification/spawnNotification', {
```

Comment:
IMO this method should return Boolean, and theme component should decide how to notify user, with notification. Another way to do this is to override this whole method in theme and it complicated further version updates